flink-runtime/src/main/java/org/apache/flink/runtime/messages/ThreadInfoSample.java

```diff
@@ -38,7 +42,7 @@ public class ThreadInfoSample implements Serializable {
 
     private ThreadInfoSample(Thread.State threadState, StackTraceElement[] stackTrace) {
         this.threadState = threadState;
-        this.stackTrace = stackTrace;
+        this.stackTrace = cleanLambdaNames(stackTrace);
```

Comment:
The `ThreadInfoSample` is common class, should we clean lambda for all ThreadInfoSamples by default? or just clean it for flame graph?

Reply:
I looked into how this class is used, and it appears that its stacktrace-related functionality is only used for flame graphs (and I think it was added for flame graphs specifically).
Doing this cleanup as close to the place where stack traces are created results in cleaner code, and I personally think it is highly unlikely for the class name change to affect anything.

Reply:
Currently, the `ThreadInfoSample ` is just used for flame graph. However, it's a common class.

The lambda name just effects the flame graph, so I prefer clean LambdaNames at the flame graph side, it's more clear, WDYT?

Reply:
Fair enough. I'll check where it can be done in the flame graph logic.